### PR TITLE
Bug/en 3244 compare hashes in verify proof

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Check if the hashes match when verifying a proof.

### CODE REVIEW

File: data/trie/patriciaMerkleTrie_test.go [hunk 1]
```diff
@@ -327,6 +327,23 @@ func emptyTrie() data.Trie {
 	return tr
 }
 
+func TestPatriciaMerkleTrie_VerifyProofFromDifferentTrieShouldNotWork(t *testing.T) {
```
Comment (raduchis) [inline data/trie/patriciaMerkleTrie_test.go]: This unit test does not test for extensionNode, branchNoce and leafNode. There is only one test and cannot test all three cases. 
Please add further tests.

File: data/trie/patriciaMerkleTrie.go [hunk 2]
```diff
@@ -163,13 +163,23 @@ func (tr *patriciaMerkleTrie) VerifyProof(proofs [][]byte, key []byte) (bool, er
 	tr.mutOperation.RLock()
 	defer tr.mutOperation.RUnlock()
 
+	wantHash, err := tr.Root()
```
Comment (raduchis) [inline data/trie/patriciaMerkleTrie.go]: There is no unit test for err here.

File: data/trie/patriciaMerkleTrie.go [hunk 3]
```diff
@@ -163,13 +163,23 @@ func (tr *patriciaMerkleTrie) VerifyProof(proofs [][]byte, key []byte) (bool, er
...
+
 	key = keyBytesToHex(key)
 	for i := range proofs {
 		encNode := proofs[i]
 		if encNode == nil {
 			return false, nil
 		}
 
+		hash := tr.hasher.Compute(string(encNode))
+		if !bytes.Equal(wantHash, hash) {
```
Comment (raduchis) [inline data/trie/patriciaMerkleTrie.go]: Are there multiple proofs for the same rootHash?

File: data/trie/patriciaMerkleTrie_test.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie_test.go]: Done.

File: data/trie/patriciaMerkleTrie.go (hunk 3 above)
Comment (raduchis) [reply data/trie/patriciaMerkleTrie.go]: nope... the wantHash changes

File: data/trie/patriciaMerkleTrie.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go]: tr.Root() is tested elsewhere, and if the wantHash is wrong, some tests will fail.

File: data/trie/patriciaMerkleTrie_test.go (hunk 1 above)
Comment (raduchis) [reply data/trie/patriciaMerkleTrie_test.go]: ok

**Pull request merged**